packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -140,11 +155,61 @@ export const treeNodeApi: TreeNodeApi = {
 
 /**
  * A collection of events that can be raised by a {@link TreeNode}.
+ *
+ * @privateRemarks
+ * TODO: add a way to subscribe to a specific field (for afterShallowChange and afterDeepChange).
+ * Probably have object node and map node specific APIs for this.
+ *
+ * TODO: ensure that subscription API for fields aligns with API for subscribing to the root.
+ *
+ * TODO: add more wider area (avoid needing tons of afterShallowChange registration) events for use-cases other than afterDeepChange.
+ * Some ideas:
+ *
+ * - afterDeepChange, but with some subtrees/fields/paths excluded
+ * - helper to batch several afterShallowChange calls to a afterDeepChange scope
+ * - parent change (ex: registration on the parent field for a specific index: maybe allow it for a range. Ex: node event takes optional field and optional index range?)
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
+	/**
+	 * Raised on a node after a change is applied to one of its fields.
+	 *
+	 * @remarks
+	 * This does not include changes to nodes within those fields:
+	 * it only includes changes of what nodes are in the fields, which field they are in, and in the case of arrays,
+	 * where the nodes are in the array.
+	 *
+	 * This does not include any changes to the location of the current node:
+	 * if a node is moved, inserted or removed, events are fired for the parent of the node, not the node itself.
+	 *
+	 * These events occur whenever the apparent contents of the node instance change, regardless of what caused the change.
+	 * For example these events will fire when the local client reassigns a child, when part of a remote edit is applied
+	 * to the node, or when the node has to be updated due to resolution of a merge conflict
+	 * (for example a previously applied local change might be undone, then reapplied differently or not at all).
+	 *
+	 * For remote edits these events are not guaranteed to occur in the same order or quantity that the edits were originally made:
+	 * While a batch of edits will get events for each change, and will as a whole update the tree to the appropriate end
+	 * state, no guarantees are made about the intermediate states other than the tree being in schema.
```

Comment:
Nit: I think we use `in-schema` to refer to things objects that are schema compliant. I think the `-` is important for clarity. Otherwise, we should say something like "the tree being compatible with the schema".
```suggestion
	 * state, no guarantees are made about the intermediate states other than the tree being in-schema.
```